### PR TITLE
AffineConstraints::compute_locally_relevant_constraints() skip step

### PR DESCRIPTION
**Pull request by peterrum**
closes https://github.com/dealii/dealii/issues/17691

@bangerth Do you mean it like this?

### CODE REVIEW

File: include/deal.II/lac/affine_constraints.templates.h [hunk 1]
```diff
@@ -587,17 +593,21 @@ AffineConstraints<number>::make_consistent_in_parallel(
...
-          locally_owned_dofs,
-          constraints_to_make_consistent,
-          mpi_communicator);
+      internal::compute_locally_relevant_constraints(
+        *this,
+        locally_owned_dofs,
+        constraints_to_make_consistent,
+        mpi_communicator,
+        iteration_count == 0,
+        imported_constraints);
```
Comment (bangerth) [inline include/deal.II/lac/affine_constraints.templates.h:610]: There is a difference here between the old and new code in that the old code *reset* `imported_constraints` to an empty vector every iteration, whereas you don't any more. I don't think you account for this anywhere in the called function. Can you undo this part of the changes? That is, go back to the function returning a vector? Let's clean that up separately.

**Review by bangerth — COMMENTED** [review]
Yes, that's pretty much exactly what I had started with as well, and that worked. Then I made more changes and did `git --amend` and made a mistake somewhere that I haven't had the time to find :-(

If you undo the change to the calling interface, I'm ok to go with this. It's exactly what I had in mind. I can then take it from there and morph it again in the direction I wanted to go with it.

File: include/deal.II/lac/affine_constraints.templates.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/lac/affine_constraints.templates.h:610]: I'm not saying that this is wrong, just that I can't see right now why it is right and I'd like to keep things incremental.

**Comment by marcfehling** [comment]
I ran this patch on my hp-Stokes scenario. It works fine. :+1:

**Review by kronbichler — APPROVED** [review]
OK with me; since @bangerth suggested to do the clean-up elsewhere, and I do not see an immediate reason why it should be tied to this case, I'd suggest to follow his suggestion.

**Pull request merged**